http-client/Network/HTTP/Client/Request.hs

```diff
@@ -127,8 +127,8 @@ applyAnyUriBasedAuth uri req =
   where
     hasAuth = (notEmpty theuser) && (notEmpty thepass)
     notEmpty = not . null
-    theuser = username authInfo
-    thepass = password authInfo
+    theuser = unEscapeString . takeWhile (/=':') . authPrefix $ authInfo
+    thepass = unEscapeString . takeWhile (/='@') . drop 1 . dropWhile (/=':') $ authInfo
```

Comment:
Because I had not notice that `username` and `password` exported here are also used to apply basic auth.
I have update my commits.